### DISCUSSION TITLE
제가 프로그램쪽을 잘 몰라 문의 드립니다

### DISCUSSION THREAD

**Discussion by Alloana** [Lastorder-DC/chatreader-kor#72] [문의] [answered]
너무 좋은 기능으로 잘 사용 하고 있습니다
사실상 채팅이 올라오는 대로 실시간 tts기능인데

이게 채팅창이 아니라
다른 창에 나타나는 문장을 실시간으로
tts변환 해주는 방법은 어려울까요?

stt로 목소리를 텍스트로 변환 시켜서
자막으로 출력되는 기능을 사용 하고 있습니다.

이때 출력되는 문장을
(웹페이지에 문장 출력후 obs에서 자막으로 나옴)

tts를 통해 실시간으로 음성으로 변환이 혹시 가능할까요?

채팅창에 올라오는 내용도 실시간 변환이 가능한데
웹페이지에 텍스트로 출력되는 문장도 음성변환이 가능한지 궁금합니다

sts(스피치 투 스피치) 기능이 필요한 셈인데
자막과 음성이 함께 나왔으면 해서 찾게되었습니다.

트위치내 채팅창인식이 아닌 다른 웹페이지 내 출력되는 문장을 인식 후 음성 출력이 많이 어려울까요?

**Comment by Lastorder-DC** [comment, marked as answer]
https://lastorder.xyz/chatreader-kor/parrot.html

크롬 브라우저 내장 STT(Speech to Text)를 이용한 페이지입니다. 인식된 음성을 TTS로 읽어주나 아직 커스텀 음성은 불가능합니다.
p.s.) <s>한글이 이상하게 들릴텐데 버그로... 수정해두겠습니다.</s> 수정되었습니다. 새로고침하시면 반영됩니다. 말을 마친뒤 조금 쉬어야 인식이 시작되며 만약 음성이 잘 인식되지 않는다면 주소창의 카메라 아이콘(아래 첨부이미지 참고)를 눌러 마이크 선택을 변경해주시면 됩니다.(지금 마이크를 연결하지 않아 스팀 가상 마이크가 인식되고 있는데 마이크 연결되어 있다면 인식될겁니다)

![2022-12-07_18-25-36](https://user-images.githubusercontent.com/18280396/206140184-713c8e5b-7228-42b0-a180-3ce3b55cfe9d.png)

**Reply by Alloana** [reply]
헉 감사함니다!

**Reply by Alloana** [reply]
여러번 테스트 하는 중인데 연속 인식 켜기를 해놔도, 몇초간 사용을 하지 않으면 자동으로 인식이 꺼져버리는데 혹시 계속 켜둘 방법은 없을까요?

**Reply by Lastorder-DC** [reply]
네. API 자체에서 꺼버리는것이라 불가능합니다.

**Reply by Alloana** [reply]
앗...감사합니다...
